python/ray/dag/constants.py

```diff
@@ -1,7 +1,14 @@
+import os
+
 # Reserved keys used to handle ClassMethodNode in Ray DAG building.
 PARENT_CLASS_NODE_KEY = "parent_class_node"
 PREV_CLASS_METHOD_CALL_KEY = "prev_class_method_call"
 BIND_INDEX_KEY = "bind_index"
 
 # Reserved key to distinguish DAGNode type and avoid collision with user dict.
 DAGNODE_TYPE_KEY = "__dag_node_type__"
+
+# Feature flag to turn off the graph verification.
+RAY_ADAG_ENABLE_VERIFY_GRAPH = (
```

Comment:
Can you call it something like DETECT_DEADLOCK? to match the function name.

Reply:
I am still using the name `RAY_ADAG_ENABLE_VERIFY_GRAPH` because I believe we will have more verification tasks in the future, not only for deadlock detection. Therefore, I use a more general name for the feature flag. I will update it to `DETECT_DEADLOCK` for now. We can add other feature flags in the future if needed.

Reply:
Update https://github.com/ray-project/ray/pull/45960/commits/f7c73d5d49bb31570bfbdf1702d5ea91aacb6a25